### PR TITLE
Added az webapp up support

### PR DESCRIPTION
**Pull request by sanchitmehta**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/appservice-kube/azext_appservice_kube/_params.py [hunk 1]
```diff
@@ -65,6 +65,19 @@ def load_arguments(self, _):
...
+        c.argument('plan', options_list=['--plan', '-p'], configured_default='appserviceplan',
+                   completer=get_resource_name_completion_list('Microsoft.Web/serverFarms'),
+                   help="name of the appserviceplan associated with the webapp")
+        c.argument('sku', arg_type=sku_arg_type)
+        c.argument('dryrun', help="show summary of the create and deploy operation instead of executing it", default=False, action='store_true')
+        c.argument('location', arg_type=get_location_type(self.cli_ctx))
+        c.argument('launch_browser', help="Launch the created app using the default browser", default=False, action='store_true', options_list=['--launch-browser', '-b'])
+        c.argument('logs', help="Configure default logging required to enable viewing log stream immediately after launching the webapp", default=False, action='store_true')
+        c.argument('html', help="Ignore app detection and deploy as an html app", default=False, action='store_true')
+        c.argument('kube_environment', help="Name of the kube environment, if app is to be deployed to a k8 cluster", options_list=['--kube-environment', '-ke'])
```
Comment (ebencarek) [inline src/appservice-kube/azext_appservice_kube/_params.py]: I think we use '-k' as the short form of the arg in ASP commands, so we should keep it consistent.

File: src/appservice-kube/azext_appservice_kube/_params.py [hunk 2]
```diff
@@ -174,5 +187,5 @@ def load_arguments(self, _):
     with self.argument_context('appservice kube show') as c:
         c.argument('name', arg_type=name_arg_type, help='Name of the Kubernetes Environment.')
 
-    with self.argument_context('appservice kube wait') as c:
+    with self.argument_context('appservice kube waite') as c:
```
Comment (ebencarek) [inline src/appservice-kube/azext_appservice_kube/_params.py]: Did you change this by accident?

File: src/appservice-kube/azext_appservice_kube/_create_util.py [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+
+
+def _resource_client_factory(cli_ctx, **_):
+    from azure.cli.core.profiles import ResourceType
+    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES)
+
+
+def web_client_factory(cli_ctx, **_):
+    from azure.mgmt.web import WebSiteManagementClient
+    return get_mgmt_service_client(cli_ctx, WebSiteManagementClient)
```
Comment (ebencarek) [inline src/appservice-kube/azext_appservice_kube/_create_util.py]: This function already exists in _client_factory.py, just import it from there. I know the method bodies are a little different but they should result in the same client being returned.

File: src/appservice-kube/azext_appservice_kube/_create_util.py [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+                         NODE_VERSIONS, NETCORE_RUNTIME_NAME, NODE_RUNTIME_NAME, DOTNET_RUNTIME_NAME,
+                         DOTNET_VERSION_DEFAULT, DOTNET_VERSIONS, STATIC_RUNTIME_NAME,
+                         PYTHON_RUNTIME_NAME, PYTHON_VERSION_DEFAULT, LINUX_SKU_DEFAULT, OS_DEFAULT)
+
+logger = get_logger(__name__)
+
+
+def _resource_client_factory(cli_ctx, **_):
+    from azure.cli.core.profiles import ResourceType
+    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES)
```
Comment (ebencarek) [inline src/appservice-kube/azext_appservice_kube/_create_util.py]: This should move to _client_factory.py

**Review by ebencarek — CHANGES_REQUESTED** [review]

File: src/appservice-kube/azext_appservice_kube/custom.py [hunk 5]
```diff
@@ -336,6 +341,180 @@ def create_webapp(cmd, resource_group_name, name, plan, runtime=None, startup_fi
...
+        parsed_id = parse_resource_id(kube_id)
+        kube_name = parsed_id.get("name")
+        kube_rg = parsed_id.get("resource_group")
+        if kube_name is not None and kube_rg is not None:
+            kube_env = cf_kube_environments(cmd.cli_ctx).get(kube_rg, kube_name)
+            if kube_env is not None and  kube_env.location is not None:
+                location = kube_env.location.replace(" ", "")
+            else:
+                raise CLIError("Kube Environment '{}' not found in subscription.".format(kube_id))
+    #AppServicePlan = cmd.get_models('AppServicePlan')
```
Comment (ebencarek) [inline src/appservice-kube/azext_appservice_kube/custom.py]: Is this line supposed to be uncommented or is it not needed?

File: src/appservice-kube/azext_appservice_kube/custom.py [hunk 6]
```diff
@@ -336,6 +341,180 @@ def create_webapp(cmd, resource_group_name, name, plan, runtime=None, startup_fi
...
+    import os
+    is_kube = False
+    KubeEnvironmentProfile = cmd.get_models('KubeEnvironmentProfile')
+    if kube_environment is not None:
+        if resource_group_name is None:
+            raise CLIError("Must provide a resource group where Kube Environment '{}' exists".format(kube_id))
+        is_kube = True
+        kube_id = _resolve_kube_environment_id(cmd.cli_ctx, kube_environment, resource_group_name)
+        kube_def = KubeEnvironmentProfile(id=kube_id)
+        kind = KUBE_ASP_KIND
```
Comment (ebencarek) [inline src/appservice-kube/azext_appservice_kube/custom.py]: It looks like this variable is unused

**Review by ebencarek — CHANGES_REQUESTED** [review]

File: src/appservice-kube/azext_appservice_kube/_params.py (hunk 1 above)
Comment (sanchitmehta) [reply src/appservice-kube/azext_appservice_kube/_params.py]: Added a flag `-e` or `--environment`, incorporating the feedback that we should keep it consistent with ASE workflow. Would have to change create flag as well.

**Pull request closed without merging**